File: frontEnd/web-chat/src/types/modelConfig.ts
export interface ModelConfig {
    id?: number | null;
    name?: string | null;
    provider: string;
    base_url?: string | null;
    chat_model: string;
    image_model?: string | null;
    api_key_masked: string;
    configured: boolean;
    is_default?: boolean;
}

export interface ModelConfigPayload {
    api_key: string;
    name?: string | null;
    base_url?: string | null;
    chat_model: string;
    image_model?: string | null;
}

File: backend/app/schemas/model_config.py
from pydantic import BaseModel


class ModelConfigRequest(BaseModel):
    api_key: str
    name: str | None = None
    base_url: str | None = None
    chat_model: str
    image_model: str | None = None


class ModelConfigResponse(BaseModel):
    id: int | None = None
    name: str | None = None
    provider: str
    base_url: str | None = None
    chat_model: str
    image_model: str | None = None
    api_key_masked: str
    configured: bool
    is_default: bool = False


class ModelConfigListResponse(BaseModel):
    configs: list[ModelConfigResponse]
